File: Problems/No.217_Contains_Duplicate/contains_duplicate.cpp
class Solution {
public:
    bool containsDuplicate(vector<int>& nums) {
        
        // parameter
        std::unordered_set<int> num_set;

        // start computing
        for (auto num: nums) {

            if (num_set.find(num) != num_set.end()) return true;
            else num_set.insert(num);

        }
        
        return false;

    }
};

File: Problems/No.217_Contains_Duplicate/contains_duplicate.py
class Solution(object):
    def containsDuplicate(self, nums):
        """
        :type nums: List[int]
        :rtype: bool
        """

        # parameter
        num_set = set()

        # start computing
        for num in nums:
            if num in num_set:
                return True
            else:
                num_set.add(num)
        
        return False
